docs/proposals/pepxxx_wheel_variant_support.md

```diff
@@ -62,88 +62,668 @@ something they [indirectly do today](https://github.com/wheelnext/wheelnext/pull
+
+The use of explicit `*-variants.json` files make it possible to publish variant wheels via indexes without explicit
+wheel variant support. The only requirement is that the index permits publishing wheels with variant-specific filenames,
+and that it permits publishing JSON files. In particular, they do not have to parse variant information in any way.
+
 
 ## Security Implications
 
-The `Provider Plugin` mechanism introduces potential security risks, such as untrusted plugins providing misleading
-platform data. To mitigate this:
+The provider plugin mechanism introduces potential security risks. Its flexibility implies that any requested package
+may introduce variant wheels, therefore causing the package manager to install arbitrary packages for the provider
+plugin, and execute code from them. This effectively goes against [the rationale for the wheel
+format](https://peps.python.org/pep-0427/#rationale), that aimed to avoid running arbitrary code at install time.
+Furthermore, it has been pointed out that packages that will be used as a regular user are installed with elevated
+privileges, effectively making arbitrary code execution even more serious of a threat.
+
+The proposal opens at least two new points in the installation supply chain for malicious actors to inject arbitrary
+code payload:
+
+1. A new version of an existing provider plugin is published with malicious code.
+
+2. A new variant is added to an existing package, with a malicious provider.
+
+Unfortunately, it is impossible to fully address these concerns without sacrificing the flexibility and user
+convenience. Nevertheless, it should be possible to reduce the risks. In particular, the following options have been
+suggested:
+
+- Providing users with more explicit control over variant usage. The package manager should provide options to disable
+  variant use entirely, allow users to specify allow-lists and block-lists for variant providers, and pin them.
+
+- Providing the ability to use a static file in place of variant provider calls. Such a file could be generated
+  by invoking the plugin in a secured environment or written manually. The installer would consult the file to get
+  the list of supported configurations rather than installing and running the plugins. Unfortunately, this will be hard
+  to maintain for dynamic plugins, since their output may differ per available variants.
+
+- Maintaining a central registry of vetted variant providers. A central authority would be responsible for scanning
+  new plugin releases for malicious code, and maintaining a list of versions and hashes of plugins confirmed
+  to be secure.
+
+- Requiring that provider plugins vendor all their dependencies, and disabling dependency installation while installing
+  them, therefore reducing the attack surface and making auditing easier.
 
-- Plugins should be sourced from trusted repositories. It's essentially remote code execution at install time.
 
 ## How to Teach This
```

Comment:
Important questions are:

* We have to ensure that users are aware that providers means running code, even when installing wheels, and that this has security implications
* How can a user of e.g. `python -m build`can get wheel variants?
* What do I, as a user, get from this extra complexity and the security implications?